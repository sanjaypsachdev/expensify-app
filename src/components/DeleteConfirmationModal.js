import React from 'react';
import ReactModal from 'react-modal';
import { connect } from 'react-redux';
import {
  startRemoveExpense,
  cancelRemoveExpense } from '../actions/expenses';

export class DeleteConfirmationModal extends React.Component {

  onRemoveExpense = () => {
    this.props.startRemoveExpense({ id: this.props.expenseToDelete.id });
    this.props.afterRemoveExpense();
  }
  
  handleCancelRemoveExpense = () => {
    this.props.cancelRemoveExpense({ id: this.props.expenseToDelete.id })
  }
  
  render() {
    return(
      <ReactModal
        ariaHideApp={false}
        isOpen={this.props.expenseToDelete.isDeleting}
        onRequestClose={this.handleCancelRemoveExpense}
        contentLabel="Confirm Deletion"
        className="modal"
      >
        <h3 className="modal__title">Are you sure you want to delete this expense ?</h3>
        <div className="modal_buttons">
          <button className="button yes" onClick={this.onRemoveExpense}>Yes</button>
          <button className="button no" onClick={this.handleCancelRemoveExpense}>No</button>
        </div>
      </ReactModal>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
  cancelRemoveExpense: (id) => dispatch(cancelRemoveExpense(id))
});

export default connect(null, mapDispatchToProps)(DeleteConfirmationModal);