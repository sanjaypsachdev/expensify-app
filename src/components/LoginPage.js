import React from 'react';
import { connect } from 'react-redux';
import { startLogin, authProviders } from '../actions/auth';

export const LoginPage = ({ startGoogleLogin, startGithubLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>It's time to get your expenses under control.</p>
      <div className="box-layout__buttons">
        <button className="button" onClick={startGoogleLogin}>Login with Google</button>
        <button className="button" onClick={startGithubLogin}>Login with Github</button>
      </div>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startGoogleLogin: () => dispatch(startLogin(authProviders.GOOGLE)),
  startGithubLogin: () => dispatch(startLogin(authProviders.GITHUB))
});

export default connect(undefined, mapDispatchToProps)(LoginPage);