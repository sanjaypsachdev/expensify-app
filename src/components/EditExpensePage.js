import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {
  startEditExpense,
  setRemoveExpense } from '../actions/expenses';
import DeleteConfirmationModal from './DeleteConfirmationModal';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }

  onSetRemoveExpense = () => {
    this.props.setRemoveExpense({ id: this.props.expense.id })
  }

  afterRemoveExpense = () => {
    this.props.history.push('/');
  }  

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.onSetRemoveExpense}>Remove Expense</button>
        </div>
        <DeleteConfirmationModal
          expenseToDelete={this.props.expense}
          afterRemoveExpense={this.afterRemoveExpense}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (expenseId, expense) => dispatch(startEditExpense(expenseId, expense)),
  setRemoveExpense: (id) => dispatch(setRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
