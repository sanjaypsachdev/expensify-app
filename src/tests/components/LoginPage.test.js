import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should render LoginPage coorectly', () => {
  const wrapper = shallow(<LoginPage startGoogleLogin={() => {  }} startGithubLogin={() => {  }} />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startGoogleLogin on button click', () => {
  const startLoginSpy = jest.fn();
  const wrapper = shallow(<LoginPage startGoogleLogin={startLoginSpy} startGithubLogin={() => {  }} />);
  wrapper.find('button').at(0).simulate('click');
  expect(startLoginSpy).toHaveBeenCalled();
});

test('should call startGithubLogin on button click', () => {
  const startLoginSpy = jest.fn();
  const wrapper = shallow(<LoginPage startGoogleLogin={() => {  }} startGithubLogin={startLoginSpy} />);
  wrapper.find('button').at(1).simulate('click');
  expect(startLoginSpy).toHaveBeenCalled();
});