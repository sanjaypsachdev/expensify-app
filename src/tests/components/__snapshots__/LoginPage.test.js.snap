// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage coorectly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control.
    </p>
    <div
      className="box-layout__buttons"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Login with Google
      </button>
      <button
        className="button"
        onClick={[Function]}
      >
        Login with Github
      </button>
    </div>
  </div>
</div>
`;
